bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/ModelRepository.java

```diff
@@ -108,6 +110,37 @@ boolean hasModel(final String modeluri) {
          });
    }
 
+   @SuppressWarnings("checkstyle:IllegalCatch")
+   public Optional<EObject> getModelElementById(final String modeluri, final String elementid) {
+      return loadResource(modeluri)
+         .flatMap(res -> {
+            try {
+               EObject modelElement = res.getEObject(elementid);
+               return Optional.ofNullable(modelElement);
+            } catch (Exception e) {
+               LOG.error("Could not load element with URI fragment: " + elementid);
+               return Optional.empty();
+            }
+         });
+   }
+
+   public Optional<EObject> getModelElementByName(final String modeluri, final String elementname) {
```

Comment:
Does it make sense to also add a test for this in ModelRepositoryTest?

Also this method will only return the FIRST element that matches the given name. That is fine by me but it should make sure that this is maybe documented somewhere. Afaik there is no real API documentation right now but if I am wrong please make sure that this behavior is documented.

Reply:
Yes this is a good point. I updated the README, where the current ModelServerClientAPI as well as the HTTP endpoints are documented. I added therefore a hint that it returns the first machting element.
Also I added a test in ModelRepositoryTest.